src/applications/vaos/api/index.js

```diff
@@ -105,12 +104,22 @@ export const getLongTermAppointmentHistory = (() => {
             .subtract(currentMonths, 'months')
             .startOf('day')
             .toISOString(),
-        ).then(newAppts => appointments.push(...newAppts));
-
+        ]);
         currentMonths += MONTH_CHUNK;
       }
 
-      promise = promise.then(() => appointments);
+      // This is weird, but hopefully clear. There are four chunks with date
+      // ranges from the array created above. We're trying to run them serially,
+      // which makes this more complicated than using Promise.all
```

Comment:
Just curious why they need to be run serially? If it's to keep them in order, I'm pretty sure Promise.all maintains the order for the results.

Reply:
Fetching two years of history in one call is too much and the request times out in staging, so I'm trying to lighten the load on the backend by not requesting all that at once, even in parallel calls. It might not be necessary, but I'm trying to be cautious because we're pulling a wide date range.

Reply:
Updated the comment to explain that better